src/components/Page/Toolbar/Toolbar.jsx

```diff
@@ -40,53 +42,61 @@ export class Toolbar extends Component {
     this._initializeModal();
   }
 
-  onBrushSizeChange = () => {
-    this.props.onBrushSizeChange(Number(this.brushSizeRangeRef.current.value));
-    this.setState({
-      brushSize: Number(this.brushSizeRangeRef.current.value)
-    })
-  }
-
-  onEraserSizeChange = () => {
-    this.props.onEraserSizeChange(Number(this.eraserSizeRangeRef.current.value));
-    this.setState({
-      eraserSize: Number(this.eraserSizeRangeRef.current.value)
-    })
+  provideRef = (tool) => {
+    if(tool === 'brush')
+      return this.brushSizeRangeRef.current.value;
+    else
+    if(tool === 'rainbow_brush')
+      return this.changeRateRangeRef.current.value;
+    else
+    if(tool === 'eraser')
+      return this.eraserSizeRangeRef.current.value;
+    else
+    if(tool === 'line')
+      return this.lineThicknessRangeRef.current.value;
   }
 
-  onLineSizeChange = () => {
-    this.props.onBrushSizeChange(Number(this.lineSizeRangeRef.current.value));
+  manageChange = (property,tool = 'brush') => {
+    let result = this.provideRef(tool);
+    this.props._changeToolSetting(property,Number(result));
     this.setState({
-      brushSize: Number(this.lineSizeRangeRef.current.value)
-    })
+      property: Number(this.colorSpeedRangeRef.current.value)
+    });
   }
 
   render() {
-    const { initialBrushColor, boardState, _setTool, _clearBoard, _save, _onUndo, _onRedo } = this.props;
+    const { initialBrushColor, boardState, _setTool, _clearBoard, _save, _onUndo, _onRedo,_changeToolSetting } = this.props;
     const [r, g, b] = initialBrushColor;
 
     return (
       <div className="toolbar">
-        <div ref={this.brushTopToolbarRef} className={`top-toolbar z-depth-1 valign-wrapper ${boardState.tool === 'brush' ? '' : 'hide'}`}>
+        <div ref={this.brushTopToolbarRef} className={`top-toolbar z-depth-1 valign-wrapper ${boardState.tool === 'brush' || 'rainbow_brush' ? '' : 'hide'}`}>
           <label>Brush Size</label>
-          <input type="range" min="2" max="100" value={this.state.brushSize} ref={this.brushSizeRangeRef} onChange={this.onBrushSizeChange} />
+          <input type="range" min="2" max="100" value={this.state.brushSize} ref={this.brushSizeRangeRef} onChange={this.manageChange('brushSize','brush')} />
+        </div>
+
+        <div ref={this.brushTopToolbarRef} className={`top-toolbar z-depth-1 valign-wrapper ${boardState.tool === 'rainbow_brush' ? '' : 'hide'}`}>
+          <label>Color Speed</label>
+          <input type="range" min="2" max="100" value={this.state.brushSize} ref={this.brushSizeRangeRef} onChange={this.manageChange('changeRate','raibow_brush')} />
```

Comment:
This uses `brushSizeRangeRef` but it is changing the `changeRate`

Reply:
😝 oops it was let there by mistake